src/service/viewport-impl.js

```diff
@@ -290,10 +296,11 @@ export class Viewport {
 
   /**
    * Returns the size of the viewport.
+   * @param {SizeReportingFidelity} fidelity
    * @return {!{width: number, height: number}}
    */
-  getSize() {
-    if (this.size_) {
+  getSize(fidelity = SizeReportingFidelity.LOW) {
+    if (this.size_ && fidelity == SizeReportingFidelity.LOW) {
```

Comment:
Oh I thought you were asking why we needed fidelity in the first place. For this one, if `this.size_` is defined but fidelity is high, we discard the old `this.size_` and recalculate anyway. if we remove fidelity then it will return the cached `this.size_` regardless of whether fidelity is high or low.